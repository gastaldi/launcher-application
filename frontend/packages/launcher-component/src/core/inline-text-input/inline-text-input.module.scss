.title {
  h1 {
    font-size: 1.6em;
    display: inline;
  }
  input {
    width: 200px;
    border: 0;
    border-bottom: 1px solid;
    &:hover,
    &:active,
    &:focus {
      outline: none;
      border-bottom-color: rgb(57, 165, 220);
    }
    background: transparent;
  }
}
