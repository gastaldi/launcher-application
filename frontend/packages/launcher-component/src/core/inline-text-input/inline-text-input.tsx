import React, { useState } from 'react';
import { Text, TextInput, TextVariants, TextInputProps } from '@patternfly/react-core';

import style from './inline-text-input.module.scss';
import { EditAltIcon } from '@patternfly/react-icons';

interface InlineTextInputProps extends TextInputProps {
  title: string;
  warning?: string;
}

export function InlineTextInput(props: InlineTextInputProps) {
  const [hint, setHint] = useState(false);
  return (
    <div className={style.title}>
      <Text component={TextVariants.h1}>{props.prefix}</Text>
      <TextInput {...props} type="text" onMouseEnter={() => setHint(!hint)} onMouseLeave={() => setHint(!hint)} />
      {hint && <EditAltIcon />}
      <span style={{display: 'block', color: '#f0ab00', paddingTop: '5px'}}>{props.warning}</span>
    </div>
  );
}
