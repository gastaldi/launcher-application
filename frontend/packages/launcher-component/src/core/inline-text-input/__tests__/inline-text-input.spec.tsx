import React from 'react';
import { render } from 'react-testing-library';

import { InlineTextInput } from '../inline-text-input';

describe('<InlineTextInput />', () => {
  it('renders the InlineTextInput correctly', () => {
    const comp = render(<InlineTextInput id="app" title="New Application:"/>);
    expect(comp.asFragment()).toMatchSnapshot();
  });
});
