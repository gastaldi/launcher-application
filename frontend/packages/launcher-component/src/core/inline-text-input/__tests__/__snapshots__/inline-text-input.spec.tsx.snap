// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InlineTextInput /> renders the InlineTextInput correctly 1`] = `
<DocumentFragment>
  <div
    class="title"
  >
    <h1
      class=""
      data-pf-content="true"
    />
    <input
      aria-invalid="false"
      class="pf-c-form-control"
      id="app"
      title="New Application:"
      type="text"
      value=""
    />
    <span
      style="display: block; color: rgb(240, 171, 0); padding-top: 5px;"
    />
  </div>
</DocumentFragment>
`;
